### ISSUE TITLE
Notification when draft is saved does not work anymore

### ISSUE THREAD

**Issue by sdebodt** [ugent-library/biblio-backend#337]
# Story
During the add publication/dataset workflow, a draft record is saved, which can then be accessed from the publications/datasets overview. However, this is not obvious to the user. Give message to user when draft is saved.

_Note; this is not a "feature parity" issue. It was available in the previous dev environment, but is now not available anymore. If this is too much work, we can put this into "nice to have"._ 

## Context
- This feature used to exist in backend.biblio.ugent.be.
  - It does not work anymore in the Eliot version.
  - The styling it used is now deprecated. Use the code below.

Question: Where there any other toast available already in the new backend?

## Success criteria
- [ ] Visible top right
- [ ] Disappears within 3200ms
- [ ] Title: Changes saved as draft.
- [ ] Body: To make your changes visible to the outside world, publish your record.
- [ ] Click on "x" to make it go away

## Component
Use the new toast component.
- Simple toast https://biblio-ugent-prototype-emily-v2.vercel.app/styleguide/b-toasts.html#component-04-b-toast-simple
- With title and body text: https://biblio-ugent-prototype-emily-v2.vercel.app/styleguide/b-toasts.html#component-02-b-toast-2-lines
- Check icon: https://biblio-ugent-prototype-emily-v2.vercel.app/styleguide/b-toasts.html#component-06-b-toast-success

Toast:
<img width="427" alt="Schermafbeelding 2022-08-29 om 20 36 21" src="https://user-images.githubusercontent.com/1572885/187273804-b791de85-5977-4679-90be-5a56b6abc282.png">

Positioning (sketch):
<img width="1440" alt="Schermafbeelding 2022-08-29 om 20 38 27" src="https://user-images.githubusercontent.com/1572885/187274176-9eb14ff7-1e87-4713-abe7-f12bed6c06eb.png">
<img width="1401" alt="Schermafbeelding 2022-08-29 om 20 39 02" src="https://user-images.githubusercontent.com/1572885/187274179-d6603520-9232-4a07-a7e5-72917275ba5a.png">

**Comment by mietcls** [comment]
Observed:

- The message appears but might disappear too fast. To discover if this message provides enough information, and if so – make it go away slower. For subtitles 2.5 seconds are the minimum. Will research what the ideal length is for contextual messages.
- The message appears bottom right. This is not common in Western applications as we read from left to right, and will see bottom right last.

Am I missing anything else here?

**Comment by mietcls** [comment]
Thank you. So we need to update the text with something in the likes of 
"Full text added and draft automatically saved."

This might also be an accessibility issue; because we take away control and expectations. We'll have to improve and see how this evolves.

**Comment by mietcls** [comment]
To be replaced with the new design for toasts from Emily. @emilyvandewalle is it possible to make the toast stay on the screen a bit longer and and position it top right?

**Comment by mietcls** [comment]
To be replaced with the new design for toasts from Emily. @emilyvandewalle is it possible to make the toast stay on the screen a bit longer and and position it top right?

**Comment by netsensei** [comment]
The "save draft" is currently a plain link to `/publication`. Adding the flash message requires a bit of extra logic. However, using the new toasts currently blocks solving this.

The new toasts aren't integrated in the overall CSS yet. Straight up replacing the old `bc-toast` displayed at the bottom right, renders the new toasts all wrong. Needs to be fixed first.

**Comment by nicolasfranck** [comment]
@netsensei as noted [here](https://github.com/ugent-library/biblio-backend/issues/593), there are several "problems" with directing to a full list after a creation wizard, even IF the record was saved.

**Comment by mietcls** [comment]
About the css:

# Suggested solution

## Height
It's probably something bootstrap applies themselves, but the flex-basis makes things look weird. We could add this:
```
.toast { 
  flex-basis: 0;
}
``` 
## Positioning
We are going to position the toast a bit different so people actually _see_ it.
We'll remove the "bottom" line.
```
#flash-messages {
  top: 2rem;
  right: 2rem;
}

```

## Z-index
There's an overall problem with toasts and help text in detail pages, not fixed here. It looks broken on detail pages, but ok on overview pages.
<img width="1440" alt="schermafbeelding_2022-09-19_om_16 42 50" src="https://user-images.githubusercontent.com/1572885/191055846-6b092943-6ccd-48c7-b5b6-69bba6003200.png">
![schermafbeelding_2022-09-19_om_16 43 16](https://user-images.githubusercontent.com/1572885/191055871-3ad48a30-ca17-487a-a45b-bc3668b89eea.png)

**Comment by netsensei** [comment]
@nicolasfranck That's a different issue. This one is about showing notifications / toasts when - but not limited to - a dataset is saved. Let's solve this one first before tackling #593.

**Comment by netsensei** [comment]
Fixed this one:

* Implements the new notification / toasts from the styleguide.
* Notification is now showing when saving a dataset or publication as a draft. (follow up on #593)
* Refactored how flash messages are defined in code.

There's two ways to create a flash message: `flash.ComplexFlash()` has a box with a header with extra info or `flash.SimpleFlash()` which is just a simple barebones box per the styleguide. I've used the SimpleFlash variant everywhere so far.

Follow up todo: localization of the displayed messages.
